core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -178,6 +178,11 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
     trimToValidSize()
   }
 
+  def closeHandler() = {
+    // File handler of the index field will be closed after the mmap is garbage collected
+    mmap = null
```

Comment:
Good point. I didn't realize we had this method. I have updated the code as suggested.